apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -225,11 +226,144 @@ type PreferredClusterSelector struct {
+	MetricSelectorGreaterThanOrEqualTo MetricSelectorOperator = "Ge"
+	// MetricSelectorEqualTo dictates Fleet to select cluster if its observed value of a given
+	// metric is equal to the values specified in the requirement.
+	MetricSelectorEqualTo MetricSelectorOperator = "Eq"
+	// MetricSelectorNotEqualTo dictates Fleet to select cluster if its observed value of a given
+	// metric is not equal to the values specified in the requirement.
+	MetricSelectorNotEqualTo MetricSelectorOperator = "Ne"
+	// MetricSelectorLessThan dictates Fleet to select cluster if its observed value of a given
+	// metric is less than the value specified in the requirement.
+	MetricSelectorLessThan MetricSelectorOperator = "Lt"
+	// MetricSelectorLessThanOrEqualTo dictates Fleet to select cluster if its observed value of a
+	// given metric is less than or equal to the value specified in the requirement.
+	MetricSelectorLessThanOrEqualTo MetricSelectorOperator = "Le"
+)
+
+// MetricSelectorRequirement is a specific metric requirement when picking clusters for
+// resource placement.
+type MetricSelectorRequirement struct {
+	// Name is the name of the metric; it should be a Kubernetes label name.
+	// +required
+	Name string `json:"name"`
+
+	// Operator specifies the relationship between a cluster's observed value of the specified
+	// metric and the values given in the requirement.
+	// +required
+	Operator MetricSelectorOperator `json:"operator"`
+
+	// Values are a list of values of the specified metric which Fleet will compare against
+	// the observed values of individual member clusters in accordance with the given
+	// operator.
+	//
+	// Each value should be a Kubernetes quantity. For more information, see
+	// https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Quantity.
+	//
+	// If the operator is Gt (greater than), Ge (greater than or equal to), Lt (less than),
+	// or `Le` (less than or equal to), Eq (equal to), or Ne (ne), exactly one value must be
+	// specified in the list.
+	//
+	// +kubebuilder:validation:MaxItems=1
+	// +required
+	Values []resource.Quantity `json:"values"`
+}
+
+// MetricSelector helps user specify metric requirements when picking clusters for resource
+// placement.
+type MetricSelector struct {
+	// MatchExpressions is an array of MetricSelectorRequirements. The requirements are AND'd.
+	// +required
+	MatchExpressions []MetricSelectorRequirement `json:"matchMetrics"`
+}
+
+type MetricSortPreference struct {
+	// Name is the name of the metric which Fleet sorts clusters by.
+	// +required
+	Name string `json:"name"`
+
+	// SortOrder explains how Fleet should perform the sort; specifically, whether Fleet should
+	// sort in ascending or descending order.
+	// +required
+	SortOrder MetricSortOrder `json:"sortOrder"`
+}
+
+type MetricSorter struct {
+	// SortByMetric is a MetricSortPreference.
 	// +required
-	LabelSelector metav1.LabelSelector `json:"labelSelector"`
+	SortByMetric *MetricSortPreference `json:"sortByMetric"`
```

Comment:
Do we expect to have more fields in this data structure? If not, why not inline MetricSortPreference?

Reply:
Hi Liqian! At this moment this data structure needs a name and an order only. Earlier you mentioned that linear interpolation might not be optimal (e.g., outliers leading to skewed scores for many clusters) so I think it might needs some additional fields to configure the interpolation behavior later (e.g., cap the max. to 80%).

Reply:
Hi Liqian! It has been updated; originally it was just wrapped in layer so that the hierarchy is consistent with the label and property (metric) selectors.